virtual_ecosystem/models/abiotic/abiotic_model.py

```diff
@@ -98,21 +98,28 @@ class AbioticModel(
         "layer_heights",
         "topofcanopy_radiation",
         "stomatal_conductance",
+        "canopy_absorption",
     ),
-    populated_by_init_vars=(
+    populated_by_init_vars=(  # TODO move functions from setup() to __init__
         "soil_temperature",
         "vapour_pressure_ref",
         "vapour_pressure_deficit_ref",
         "air_temperature",
         "relative_humidity",
-        "vapour_pressure",
         "vapour_pressure_deficit",
         "atmospheric_pressure",
         "atmospheric_co2",
+        "canopy_absorption",  # DAVID This is assuming that abiotic runs before plants
```

Comment:
@davidorme can you please comment on this?

Reply:
Yeah  - this should be a `required_init_vars` instead and `plants` will have it as `populated_by_init_vars`